### PR TITLE
feat: upload/download smoke test

### PR DESCRIPTION
**Pull request by mrekucci**

### CODE REVIEW

File: pkg/check/smoke/smoke.go [hunk 1]
```diff
@@ -3,136 +3,132 @@ package smoke
...
 // NewDefaultOptions returns new default options
 func NewDefaultOptions() Options {
 	return Options{
-		Bytes:           0,
-		NodeGroup:       "bee",
-		Runs:            1,
-		Seed:            0,
-		Timeout:         1 * time.Second,
-		UploadNodeCount: 1,
+		ContentSize:   5 * 1024 << 10,
```
Comment (istae) [inline pkg/check/smoke/smoke.go:33]: can we make this just 5000000 bytes

File: pkg/check/smoke/smoke.go [hunk 2]
```diff
@@ -3,136 +3,132 @@ package smoke
...
+		Iterations:    1,
+		RndSeed:       0,
+		GasPrice:      "",
+		PostageAmount: 1000,
+		PostageDepth:  16,
+		PostageLabel:  "test-label",
+		SyncTimeout:   time.Second,
```
Comment (istae) [inline pkg/check/smoke/smoke.go:40]: this is too short I believe, let's make this like 1m

File: pkg/check/smoke/smoke.go (hunk 1 above)
Comment (mrekucci) [reply pkg/check/smoke/smoke.go:33]: Done.

File: pkg/check/smoke/smoke.go (hunk 2 above)
Comment (mrekucci) [reply pkg/check/smoke/smoke.go:40]: Done.

**Review by acud — COMMENTED** [review]
1. long running test
2. histogram for timing metrics
3. bytes endpoint
4. change postage amount to 1mil, depth 20

**Review by istae — APPROVED** [review]

**Comment by mrekucci** [comment]
>     1. long running test
> 
>     2. histogram for timing metrics
> 
>     3. bytes endpoint
> 
>     4. change postage amount to 1mil, depth 20

Done.

File: pkg/check/smoke/metrics.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+		),
+		FileDownloadErrors: prometheus.NewCounter(
+			prometheus.CounterOpts{
+				Namespace: m.Namespace,
+				Subsystem: subsystem,
+				Name:      "download_errors_count",
+				Help:      "The total number of errors encountered before successful download.",
+			},
+		),
+		FileUploadDuration: prometheus.NewHistogram(
```
Comment (acud) [inline pkg/check/smoke/metrics.go:43]: consider stripping `File` prefix from metrics

File: pkg/check/smoke/smoke.go [hunk 4]
```diff
@@ -3,136 +3,205 @@ package smoke
...
-		}
+		perm := rnd.Perm(cluster.Size())
+		txIdx := perm[0]
+		rxIdx := perm[1]
 
-		addr, err := uClient.UploadBytes(ctx, data, api.UploadOptions{Pin: false, Tag: tr.Uid})
-		if err != nil {
-			return fmt.Errorf("upload to node %s: %w", nodeName, err)
+		if txIdx == rxIdx {
+			fmt.Println("warning: uploading node is the same as downloading node!")
```
Comment (acud) [inline pkg/check/smoke/smoke.go:100]: why not `continue`?

File: pkg/check/smoke/smoke.go [hunk 5]
```diff
@@ -3,136 +3,205 @@ package smoke
...
+				return nil
+			default:
+			}
+
+			address, txDuration, err = test.upload(txName, txData)
+			if err != nil {
+				txErrors++
+				c.metrics.FileUploadErrors.Inc()
+				fmt.Printf("upload failed: %v\n", err)
+				fmt.Printf("retraing in: %v", o.TxOnErrWait)
```
Comment (acud) [inline pkg/check/smoke/smoke.go:130]: retrying

File: pkg/check/smoke/smoke.go [hunk 6]
```diff
@@ -3,136 +3,205 @@ package smoke
...
+				return nil
+			default:
+			}
+
+			rxData, rxDuration, err = test.download(rxName, address)
+			if err != nil {
+				rxErrors++
+				c.metrics.FileDownloadErrors.Inc()
+				fmt.Printf("download failed: %v\n", err)
+				fmt.Printf("retraing in: %v", o.RxOnErrWait)
```
Comment (acud) [inline pkg/check/smoke/smoke.go:149]: retrying

File: pkg/check/smoke/smoke.go [hunk 7]
```diff
@@ -3,136 +3,205 @@ package smoke
...
 		return err
 	}
 
+	time.Sleep(5 * time.Second) // Wait for the nodes to warmup.
+
 	var (
-		rnd         = random.PseudoGenerator(o.Seed)
-		r           = rand.New(rand.NewSource(o.Seed))
-		sortedNodes = ng.NodesSorted()
+		txErrors int
```
Comment (acud) [inline pkg/check/smoke/smoke.go:75]: these are incremented but never read

**Review by acud — COMMENTED** [review]
Looks good! I have a few minor comments

File: pkg/check/smoke/metrics.go (hunk 3 above)
Comment (mrekucci) [reply pkg/check/smoke/metrics.go]: This was a leftover when refactoring from `/bzz` to `/bytes`. Good catch!

File: pkg/check/smoke/smoke.go (hunk 4 above)
Comment (mrekucci) [reply pkg/check/smoke/smoke.go]: Done.

File: pkg/check/smoke/smoke.go (hunk 5 above)
Comment (mrekucci) [reply pkg/check/smoke/smoke.go]: Done.

File: pkg/check/smoke/smoke.go (hunk 6 above)
Comment (mrekucci) [reply pkg/check/smoke/smoke.go]: Done.

File: pkg/check/smoke/smoke.go (hunk 7 above)
Comment (mrekucci) [reply pkg/check/smoke/smoke.go]: Removed.

**Review by acud — APPROVED** [review]
Nice!

File: config/mainnet.yaml [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+# node-groups may inherit it's configuration from already defined node-group and override specific fields from it
+node-groups:
+  mainnet:
+    _inherit: "default"
+    persistence-enabled: true
+
+# bee-configs defines Bee configuration that can be assigned to node-groups
+# bee-configs may inherit it's configuration from already defined bee-config and override specific fields from it
+bee-configs:
+  testnet:
```
Comment (acud) [inline config/mainnet.yaml:35]: seems like this is defined twice (once below too)

File: config/staging.yaml [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+# node-groups may inherit it's configuration from already defined node-group and override specific fields from it
+node-groups:
+  staging:
+    _inherit: "default"
+    persistence-enabled: true
+
+# bee-configs defines Bee configuration that can be assigned to node-groups
+# bee-configs may inherit it's configuration from already defined bee-config and override specific fields from it
+bee-configs:
+  testnet:
```
Comment (acud) [inline config/staging.yaml:35]: not sure if it is needed

File: pkg/check/smoke/smoke.go [hunk 10]
```diff
@@ -3,136 +3,200 @@ package smoke
...
-	for i := 0; i < o.Runs; i++ {
-		uploader := r.Intn(len(sortedNodes))
-		nodeName := sortedNodes[uploader]
-		uClient, err := ng.NodeClient(nodeName)
-		if err != nil {
-			return err
-		}
-
-		fmt.Printf("run %d, uploader node is: %s\n", i, nodeName)
+	time.Sleep(5 * time.Second) // Wait for the nodes to warmup.
```
Comment (acud) [inline pkg/check/smoke/smoke.go:72]: nit, not sure if it is needed

**Review by acud — APPROVED** [review]

File: config/mainnet.yaml (hunk 8 above)
Comment (mrekucci) [reply config/mainnet.yaml:35]: Removed.

File: config/staging.yaml (hunk 9 above)
Comment (mrekucci) [reply config/staging.yaml:35]: Removed.

**Pull request merged**